src/web/demo.rkt

```diff
@@ -367,21 +377,27 @@
      (redirect-to (add-prefix (format "~a.~a/graph.html" hash *herbie-commit*)) see-other))
    (url main)))
 
+(define (run-sample hash formula)
+  (hash-set! *jobs* hash (*timeline*))
+  (define sema (make-semaphore))
+  (thread-send *worker-thread* (list 'sample hash formula sema))
+  sema)
+
 ; /api/sample endpoint: test in console on demo page:
 ;; (await fetch('/api/sample', {method: 'POST', body: JSON.stringify({formula: "(FPCore (x) (- (sqrt (+ x 1))))", seed: 5})})).json()
 (define sample-endpoint
   (post-with-json-response
     (lambda (post-data)
-      (define formula (read-syntax 'web (open-input-string (hash-ref post-data 'formula))))
+      (define formula-str (hash-ref post-data 'formula))
+      (define formula (read-syntax 'web (open-input-string formula-str)))
+      (define hash (sha1 (open-input-string formula-str)))
+      ;; Hmm how should I pass seed?
```

Comment:
Figured out I can just pass the seed in along with the list of other parameters/arguments.

Reply:
Oh this is a good point, yeah, we probably want to pass the seed in the message instead of using the pre-set one.